portal-bridge/src/types/state/trie_walker.rs

```diff
@@ -0,0 +1,159 @@
+use alloy_primitives::{Bytes, B256};
+use anyhow::ensure;
+use eth_trie::{decode_node, node::Node};
+use hashbrown::HashMap as BrownHashMap;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct AccountProof {
+    pub path: Vec<u8>,
+    pub proof: Vec<Bytes>,
+}
+
+pub struct ParentWithPath {
+    pub parent_hash: B256,
+    pub path_nibbles: Vec<u8>,
+}
+
+impl ParentWithPath {
+    pub fn new(parent_hash: B256, path_nibbles: Vec<u8>) -> Self {
+        Self {
+            parent_hash,
+            path_nibbles,
+        }
+    }
+}
+
+/// This struct takes in a root hash and a hashmap of changed nodes, then you can call an iterator
+/// which will return every proof to gossip
+pub struct TrieWalker {
+    pub nodes: BrownHashMap<B256, Vec<u8>>,
+    paths_to_parent: BrownHashMap<B256, ParentWithPath>,
```

Comment:
nit: Since we renamed the type, I would also rename this field (and everywhere below, including `get_paths_backwards` function).

I would say just `parents` is fine, but `parent_with_path_map` or something similar is also ok.